api/src/main/java/org/openmrs/module/ugandaemrreports/reports2019/SetupMaternityRegister2019.java

```diff
@@ -0,0 +1,272 @@
+import org.openmrs.module.ugandaemrreports.reporting.metadata.Dictionary;
+import org.openmrs.module.ugandaemrreports.reports.UgandaEMRDataExportManager;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+
+import java.util.ArrayList;
+import java.util.Date;
+import java.util.List;
+import java.util.Properties;
+
+/**
+ * Integrated Maternity Register Report
+ */
+
+@Component
+public class SetupMaternityRegister2019 extends UgandaEMRDataExportManager {
+
+
+	@Autowired
+	private DataFactory df;
+	@Autowired
+	SharedDataDefintion sdd;
+
+	@Autowired
+	BasePatientDataLibrary patientDataDefinition;
+
+	@Override
+	public String getExcelDesignUuid() {
+		return "28ee759f-4da7-46f6-8fff-7e1ad795660e ";
+	}
+
+
+	@Override
+	public String getDescription() {
+		return "Integrated Maternity Register 2019";
+	}
+
+	@Override
+	public String getName() {
+		return "Integrated Maternity Register 2019";
+	}
+
+	@Override
+	public String getUuid() {
+		return "694cb788-c3d1-406a-be2d-65e322f71205";
+	}
+
+	@Override
+	public String getVersion() {
+		return "1.7";
+	}
+
+	/**
+	 * @return the uuid for the report design for exporting to Excel
+	 */
+
+	@Override
+	public List<Parameter> getParameters() {
+		List<Parameter> l = new ArrayList<Parameter>();
+		l.add(df.getStartDateParameter());
+		l.add(df.getEndDateParameter());
+
+		return l;
+	}
+
+	@Override
+	public List<ReportDesign> constructReportDesigns(ReportDefinition reportDefinition) {
+		List<ReportDesign> l = new ArrayList<ReportDesign>();
+		l.add(buildExcel(reportDefinition));
+		l.add(buildReportDesign(reportDefinition));
+		return l;
+	}
+
+	/**
+	 * Build the report design for the specified report, this allows a user to override the report
+	 * design by adding properties and other metadata to the report design
+	 *
+	 * @param reportDefinition
+	 * @return The report design
+	 */
+	@Override
+	public ReportDesign buildReportDesign(ReportDefinition reportDefinition) {
+		ReportDesign rd = createCSVDesign(getExcelDesignUuid(), reportDefinition);
+		return rd;
+	}
+
+
+	public ReportDesign buildExcel(ReportDefinition reportDefinition) {
+		ReportDesign rd = createExcelTemplateDesign("8d271abd-e55d-4e5c-a9a9-495b6115fec4", reportDefinition, "MaternityRegister2019.xls");
+
+		Properties props = new Properties();
+		props.put("repeatingSections", "sheet:1,row:8-12,dataset:Maternity");
+		props.put("sortWeight", "5000");
+		rd.setProperties(props);
+		return rd;
+	}
+
+	@Override
+	public ReportDefinition constructReportDefinition() {
+
+		ReportDefinition rd = new ReportDefinition();
+		rd.setUuid(getUuid());
+		rd.setName(getName());
+		rd.setDescription(getDescription());
+		rd.setParameters(getParameters());
+		rd.addDataSetDefinition("Maternity", Mapped.mapStraightThrough(dataSetDefinition()));
+		return rd;
+	}
+
+
+
+
+	private Concept getConcept(String uuid) {
```

Comment:
@solemabrothers Global this should be in a global location - use the DataFactory I think